rust/private/rust.bzl

```diff
@@ -1139,6 +1139,89 @@ rust_test_binary = rule(
         """),
 )
 
+def rust_test_suite(name, srcs, **kwargs):
+    """A rule for creating a test suite for a set of `rust_test` targets.
+
+    This rule can be used for setting up typical rust [integration tests][it]. Given the following
+    directory structure:
+
+    ```text
+    [crate]/
+        BUILD.bazel
+        src/
+            lib.rs
+            main.rs
+        tests/
+            integrated_test_a.rs
+            integrated_test_b.rs
+            integrated_test_c.rs
+            patterns/
+                fibonacci_test.rs
+    ```
+
+    The rule can be used to generate [rust_test](#rust_test) targets for each source file under `tests`
+    and a [test_suite][ts] which encapsulates all tests.
+
+    ```python
+    load("//rust:defs.bzl", "rust_binary", "rust_library", "rust_test_suite")
+
+    rust_library(
+        name = "math_lib",
+        srcs = ["src/lib.rs"],
+    )
+
+    rust_binary(
+        name = "math_bin",
+        srcs = ["src/main.rs"],
+    )
+
+    rust_test_suite(
+        name = "integrated_tests_suite",
+        srcs = glob(["tests/**"]),
+        deps = [":math_lib"],
+    )
+    ```
+
+    [it]: https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html
+    [ts]: https://docs.bazel.build/versions/master/be/general.html#test_suite
+
+    Args:
+        name (str): The name of the `test_suite`.
+        srcs (list): All test sources, typically `glob(["tests/**/*.rs"])`.
+        **kwargs (dict): Additional keyword arguments. Args from [rust_test](#rust_test) are parsed out and shared
+            between all tests while [common attributes](https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes)
+            are shared between the test targets and `test_suite`.
+    """
+    tests = []
+
+    # Gather any rust attributes
```

Comment:
I don't think we should be filtering these and trying to treat them specially - I think we should add all attributes to the underlying `rust_test`, and call out anything we put on the `test_suite` explicitly. If someone wants to do something like set `tags` on the tests, I think that's reasonable to let them do...

Reply:
I think I've updated this accordingly. Let me know